Implements generator script and data loader script

ecuyle opened the pull request:
@Cowie22 
Please review the following PR:
Commit c8d5c31: Update README formatting for consistency

Commit 0bf2b35: Implement generator script
Generator creates 10M records for both Reservations and Restaurants
table.

Commit 80d6012: Implement PostgreSQL data loader
Loader performs batch COPY from a reservations.csv and restaurants.csv
file generated from /generator/index.js into the respective tables.
Be sure to chmod appropriate permissions before running dbSetup.sh.

ecuyle committed:
Update README formatting for consistency

ecuyle committed:
Implement generator script

Generator creates 10M records for both Reservations and Restaurants
table.

ecuyle committed:
Implement PostgreSQL data loader

Loader performs batch COPY from a reservations.csv and restaurants.csv
file generated from /generator/index.js into the respective tables.
Be sure to chmod appropriate permissions before running dbSetup.sh.

Cowie22 reviewed (approved):
Look great!  Nice work!

ecuyle committed:
Implement DataGenerator class

DataGenerator class is the new implementation used to generate
production-grade data sets. The .generate method, when using
a CHUNK_SIZE of 20,000 and MAX_ROWS count of 10,000,000, can generate
two .csv files of 10,000,000 lines each in roughly 49 seconds whilst
only consuming roughly 60mb of RAM in the process. This was achieved
by implementing both a Readable and Writable stream to reduce the load
on memory at any given point in time.

This is a huge improvement from before as the solution using only
Writable streams took roughly 90 seconds to run and consumed roughly
1.5GB of RAM in the process.
